### PR TITLE
[build-system] Added support for compiling Swift with PGO (IR) support.

### PR DESCRIPTION
**Pull request by Rostepher**
# Purpose

This PR adds support in Swift's build system to compile `swiftc` with LLVM profile generation tooling enabled and also to build Swift using a generated profile.

rdar://21800776

### CODE REVIEW

**Comment by bob-wilson** [comment]
Do you have any performance measurements to show if this helps?

File: CMakeLists.txt [hunk 1]
```diff
@@ -860,7 +881,73 @@ if(NOT DEFINED SWIFT_API_NOTES_INPUTS)
...
 
+#
+# Configure Swift to build with profile generation
+#
+
+if(SWIFT_ENABLE_PROFILE_GENERATION OR SWIFT_ENABLE_IR_PROFILE_GENERATION)
+  if(NOT SWIFT_PROFILE_MERGE_POOL_SIZE)
+    # A pool size of 1-2 is probably sufficient on a SSD. 3-4 should be fine
+    # for spining disks. Anything higher may only help on slower mediums.
+    set(SWIFT_PROFILE_MERGE_POOL_SIZE "4")
```
Comment (lplarson) [inline CMakeLists.txt]: Do you have any concrete numbers that describe how the size of the merge pool affects build times?

**Review by lplarson — APPROVED** [review]
Overall, LGTM.

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by Rostepher** [comment]
Closing stale PR.

**Pull request closed without merging**